### PR TITLE
Nobody is assigned to any of the declarations generated by the data generator  || Orcvs 3684

### PR DESCRIPTION
**Pull request by modupeadeonojobi**
This command "REGISTER=false CERTIFY=false yarn data-generator" assign declarations to nobody while "yarn data-generator" assign declarations to Kennedy Mweene.
fara-dg

### CODE REVIEW

**Review by rikukissa — APPROVED** [review]
👍

File: src/data-generator/index.ts [hunk 1]
```diff
@@ -651,91 +657,99 @@ function birthDeclarationWorkflow(
...
-              days: 1
-            }),
-            location,
-            declaration
-          )
-        } catch (error) {
-          console.log(error)
-          console.log(JSON.stringify(declaration))
-          throw error
+        if (REGISTER && CERTIFY) {
```
Comment (Zangetsu101) [inline src/data-generator/index.ts]: If someone wants to proceed till `CERTIFY` i.e. only uses `CERTIFY=false` then the declaration will not be downloaded from here and down the line when trying to reject the declaration, it should throw an error as the declaration was not downloaded.

File: src/data-generator/index.ts [hunk 2]
```diff
@@ -651,91 +657,99 @@ function birthDeclarationWorkflow(
...
+          }
+          await fetchRegistration(randomRegistrar, id)
+          if (!declaredRecently || Math.random() > 0.5) {
+            await markAsRegistered(randomRegistrar, id, registrationDetails)
+            const registration = await fetchRegistration(randomRegistrar, id)
+            if (
+              CERTIFY &&
+              (!declaredRecently || Math.random() > 0.5) &&
+              registration
+            ) {
```
Comment (Zangetsu101) [inline src/data-generator/index.ts]: The declaration is still being downloaded even if `CERTIFY` is false

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]
The changes need to be made on `deathDeclarationWorkflow` as well

**Comment by Zangetsu101** [comment]
Closing in favor of #356

**Pull request closed without merging**